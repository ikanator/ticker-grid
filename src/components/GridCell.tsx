import React from 'react';
import styled from 'styled-components';
import { color, ColorProps, space, SpaceProps } from 'styled-system';

type GridCellProps = {
  filled: boolean;
};

interface GridProps extends ColorProps, SpaceProps {}

const GridCellStyled = styled.td<GridProps>`
  ${color};
  ${space};
  border: ${({ theme }) => `${theme.space.sm}px solid ${theme.colors.common.secondary}`};
`;

export const GridCell: React.FC<GridCellProps> = ({ filled }) => (
  <GridCellStyled backgroundColor={filled ? 'common.filled' : 'common.empty'} padding="md" />
);
