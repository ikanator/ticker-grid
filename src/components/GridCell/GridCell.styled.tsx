import styled from 'styled-components';
import { color, space, ColorProps, SpaceProps } from 'styled-system';

interface GridCellStyledProps extends ColorProps, SpaceProps {}

export const GridCellStyled = styled.td<GridCellStyledProps>`
  ${color};
  ${space};
  border: ${({ theme }) => `${theme.space.sm}px solid ${theme.colors.common.secondary}`};
`;
