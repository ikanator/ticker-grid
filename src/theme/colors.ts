export interface ThemeColors {
  primary: string;
  secondary: string;
}

export const colors: ThemeColors = {
  primary: '#000',
  secondary: '#fff',
};
