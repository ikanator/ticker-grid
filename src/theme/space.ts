export interface Space {
  none: number;
  sm: number;
  md: number;
  lg: number;
}

// 0, 2, 4, 8... is preferred, but used custom ones here
export const space: Space = {
  none: 0,
  sm: 2,
  md: 6,
  lg: 12,
};
