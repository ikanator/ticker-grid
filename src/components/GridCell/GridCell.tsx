import React from 'react';
import { GridCellStyled } from './GridCell.styled';

type GridCellProps = {
  filled: boolean;
};

export const GridCell: React.FC<GridCellProps> = ({ filled }) => (
  <GridCellStyled backgroundColor={filled ? 'common.filled' : 'common.empty'} padding="md" />
);
