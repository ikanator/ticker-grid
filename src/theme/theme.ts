import { DefaultTheme } from 'styled-components';
import { colors } from './colors';
import { space } from './space';

export const defaultTheme: DefaultTheme = {
  space: { ...space },
  colors: { ...colors },
  // TODO: add breakpoints, typography, etc.
};
