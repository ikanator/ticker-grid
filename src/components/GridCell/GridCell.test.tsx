import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '../../theme/theme';
import { GridCell } from './GridCell';

describe('GridCell', () => {
  it('should render empty cell', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <table>
          <tbody>
            <tr>
              <GridCell filled={false} />
              <GridCell filled />
            </tr>
          </tbody>
        </table>
      </ThemeProvider>
    );

    const cells = Array.from(container.querySelectorAll('td'));

    expect(cells[0]).toHaveStyle('background-color: #fff;');
    expect(cells[1]).toHaveStyle('background-color: #000;');
    expect(container).toMatchSnapshot();
  });
});
