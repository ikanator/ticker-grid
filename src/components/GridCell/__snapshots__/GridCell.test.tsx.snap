// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCell should render empty cell 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="sc-bdfBQB iPCxOh"
        />
        <td
          class="sc-bdfBQB bANFBv"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
